lib/modules/actions.js

```diff
@@ -54,7 +56,7 @@ export const reroutePage = () => async (dispatch, getState) => {
       queryParams: currentPage.queryParams,
       hashParams: currentPage.hashParams,
       bodyParams: {},
-      referrer: currentPage.referrer,
+      referrer: currentPage.referrer || NO_REFERRER,
```

Comment:
This problem is solvable without introducing more data to be sent over the wire. We know two things:

1. The initial page request may or may not have a referrer
2. Subsequent page requests should always have a referrer

Therefore, a simpler solution would be to just check the history and grab the domain of the last page state (if we are not on the initial navigation).